lib/oli/delivery/metrics.ex

```diff
@@ -1324,7 +1326,12 @@ defmodule Oli.Delivery.Metrics do
   # the level of proficiency for that container.
   # The proficiency is calculated by finding the mode where the highest number of
   # users fall into a proficiency range for that container.
-  defp bucket_into_container_mode(page_data, contained_pages) do
+  defp bucket_into_container_mode(page_data, contained_pages, section_slug) do
+    student_ids =
```

Comment:
If all that this implementation needs is the set of user ids of enrolled students, it should execute a query *exactly* for that - as opposed to fetching all of the user records and filtering and mapping.   User records can be arbitrarily large, thanks to the `state` field, which is used by extrinsic storage.

Reply:
Good point. Change applied